l10n_ar_sale/models/sale_order.py

```diff
@@ -62,26 +62,36 @@ def _get_sale_checkbook(self):
 
     @api.model_create_multi
     def create(self, vals):
+        """ Elegimos la secuencia que va usar el pedido de ventas, Si la orden de venta se crea por interfaz funciona
+        ok, pero cuando viene de otro lado tipo creado por ML parece que no todos los campos o datos estan llenos o
+        se calculan como deberian, Para evitar que suceda eso seguimos esta logica:
+
+        1. Si usa talonario de ventas
+            1.1 Si tiene talonario de ventasy tiene secuencia usa esa
+            1.2 Si el anterior no se cump,e y usa tipo de venta y dentro hay un talonario con secuencia usa esa.
+        2. Si no se cumplen las anteriores y tiene tipo de venta intenta obtener si tiene secuencia.
+        3. Si no consigue secuencia deja el name como New y utiliza la secuencia por defecto para las ventas """
+
         for val in vals:
             if val.get('name', _('New')) == _('New'):
-                sale_checkbook_id = val.get('sale_checkbook_id')
-                sale_order_type = val.get('type_id')
+                sale_checkbook = self.env["sale.checkbook"].browse(val.get('sale_checkbook_id')) \
+                    if val.get('sale_checkbook_id') else False
+                sale_order_type = self.env["sale.order.type"].browse(val.get('type_id')) \
+                    if val.get('type_id') else False
 
                 use_checkbook = self.sudo().env.ref('l10n_ar_sale.use_sale_checkbook').users
-                object_model = False
-                object_id = False
-                if use_checkbook and sale_checkbook_id:
-                    object_model = 'sale.checkbook'
-                    object_id = sale_checkbook_id
-                elif (use_checkbook and not sale_checkbook_id and sale_order_type) and \
-                     (not use_checkbook and sale_order_type):
-                    object_model = 'sale.order.type'
-                    object_id = sale_order_type
-
-                if object_model and object_id:
-                    object_record = self.env[object_model].browse(object_id)
-                    val['name'] = object_record.sequence_id and \
-                        object_record.sequence_id._next() or _('New')
+
+                seq = False
+                if use_checkbook:
+                    if sale_checkbook:
+                        seq = sale_checkbook.sequence_id
+                    if not seq and sale_order_type and sale_order_type.sale_checkbook_id:
+                        seq = sale_order_type.sale_checkbook_id.sequence_id
+                if not seq and sale_order_type:
```

Comment:
```suggestion
                if not seq and sale_checkbook:
                      seq = sale_checkbook.sequence_id
                if not seq and sale_order_type:
```
¿faltaría agregar esto?